src/simplnx/DataStructure/DataObject.hpp

```diff
@@ -205,7 +205,7 @@ class SIMPLNX_EXPORT DataObject
    * @brief Checks and returns if the DataObject can be renamed to the provided
    * value.
    * @param name
-   * @return bool
+   * @return int: 0 = false, 1 = true, 2 = duplicate object found
```

Comment:
```suggestion
   * @return bool
```

If you are going to return more than true or false, do not use bool return type but perhaps a uint8 or enum.

I would expect a duplicate object to default to failing since overwriting the existing value is not the default and never has been. I am fairly certain an int value of 2 is still true. Adding an optional parameter would add clarity on when this is expected.

However, since DataObject::canRename is not changed in the .cpp file, it would be best to just reset the `@return` description to what it was. There are no 0, 1, and 2 special return values, only true and false.